### ISSUE TITLE
[clang-tidy][false negative] bugprone-implicit-widening-of-multiplication-result not detected inside std::make_unique

### ISSUE THREAD

**Issue by hexagonrecursion** [llvm/llvm-project#110960]
[godbolt](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:1,endLineNumber:12,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:12,startColumn:1,startLineNumber:1),source:'%23include+%3Cmemory%3E%0A%0Aclass+C+%7B%0Apublic:%0A++++C(long)+%7B%7D%0A%7D%3B%0A%0Aint+main()+%7B%0A++++int+x+%3D+0%3B%0A++++std::make_unique%3CC%3E(x+*+x)%3B%0A%7D%0A'),l:'5',n:'1',o:'C%2B%2B+source+%231',t:'0'),(h:compiler,i:(compiler:clang_trunk,filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',debugCalls:'1',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1',verboseDemangling:'0'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',overrides:!(),selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+clang+(trunk)+(Editor+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0'),(g:!((h:tool,i:(args:'--checks%3D!'-*,bugprone-implicit-widening-of-multiplication-result!'',argsPanelShown:'0',compilerName:'x86-64+clang+(trunk)',editorid:1,fontScale:14,fontUsePx:'0',j:1,monacoEditorHasBeenAutoOpened:'1',monacoEditorOpen:'1',monacoStdin:'1',stdin:'',stdinPanelShown:'1',toolId:clangtidytrunk,wrap:'1'),l:'5',n:'0',o:'clang-tidy+(trunk)+x86-64+clang+(trunk)+(Editor+%231,+Compiler+%231)',t:'0')),k:50,l:'4',n:'0',o:'',s:0,t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

```c++
#include <memory>

class C {
public:
    C(long) {}
};

int main() {
    int x = 0;
    std::make_unique<C>(x * x);  // should warn here, but does not
}
```

**Comment by chrchr-github** [comment]
Reduced:
~~~c++
template<typename T, typename... Args>
T f(Args&&... args) { return T(args...); }

class C {
public:
    C(long) {}
};

int main() {
    int x = 0;
    f<C>(x * x);
}
~~~
https://godbolt.org/z/fh7Efh9oM
